dashboard/lib/services/curriculum_pdfs/resources.rb

```diff
@@ -155,6 +155,15 @@ def generate_lesson_resources_title_page(lesson, directory = "/tmp/")
           end
           raise "File #{path.inspect} does not exist after generation" unless File.exist?(path)
 
+          if DCDO.get('use-ghostscript-to-generate-pdfs', false)
+            # Regenerate the PDF using Ghostscript
+            optimized_path = File.join(directory, "optimized_#{filename}")
+            gs_command = "gs -q -sDEVICE=pdfwrite -dNOPAUSE -dBATCH -sOutputFile=#{optimized_path} #{path}"
+            system(gs_command)
+            return optimized_path
```

Comment:
I get that we want to keep this fix scoped to just the curriculum pdfs for now, but I'd still like to keep all of our manual system invocations of `gs` isolated to [`collate.rb`](https://github.com/code-dot-org/code-dot-org/blob/staging/lib/pdf/collate.rb). Can we add a new helper method to that class that can be called here? Something like `PDF.regenerate_with_ghostscript`, `PDF.regenerate_embedded_font_names`, or even just `PDF.normalize_for_merging`. We can then add a nice verbose comment to that method explaining all the context we uncovered about why this is necessary!